lib/routes/bins/run.js

```diff
@@ -1,47 +1,61 @@
 const debug = require("debug")("mockbin");
 
-module.exports = (client) => (req, res, next) => {
-	// compoundId allows us to provide paths in the id to resolve to a specific bin
-	const compoundId = req.params.uuid + req.params[0];
-	client.get(`bin:${compoundId}`, function (err, value) {
-		if (err) {
-			debug(err);
-
-			throw err;
+function runMock(err, value, req, res, client, compoundId, next) {
+	if (err) {
+		debug(err);
+
+		throw err;
+	}
+
+	if (value) {
+		const har = JSON.parse(value);
+
+		// log interaction & send the appropriate response based on HAR
+		client.rpush(`log:${compoundId}`, JSON.stringify(req.har.log.entries[0]));
+		client.ltrim(`log:${compoundId}`, 0, 100);
+
+		// headers
+		har.headers.forEach((header) => {
+			res.set(header.name.trim(), header.value);
+		});
+
+		// cookies
+		har.cookies.forEach((cookie) => {
```

Comment:
Should we be setting har.cookies to default to an empty array in case it's not provided?